### ISSUE TITLE
Serenity generates green report even if tests are failed

### ISSUE THREAD

**Issue by alexandrulup** [serenity-bdd/serenity-core#1482]
serenity-core-version : 2.0.20
I'm running data-driven tests with SerenityParameterizedRunner.class and even if the results are failed, I have a green pie chart

**Comment by wakaleo** [comment]
I can't reproduce this, can you provide a sample project?

**Comment by andreiaionitoaiesv** [comment]
Sample pom :
https://gist.github.com/andreiaionitoaiesv/4095b12cb633a4aa6d672a06399ca6ff
Sample test:
https://gist.github.com/andreiaionitoaiesv/18373db4040af50f0a61efdc371e5417

![failure](https://user-images.githubusercontent.com/14057245/49863343-c4d09e80-fe08-11e8-8edb-5358e611bf3f.PNG)
![landingpage](https://user-images.githubusercontent.com/14057245/49863344-c4d09e80-fe08-11e8-8a6a-ee0dbadf7727.PNG)

**Comment by wakaleo** [comment]
Sorry, still can't reproduce based on the examples you have given. Can you provide a full sample project that demonstrates the issue?

**Comment by vuthelinh** [comment]
@alexandrulup 
I think because the result is override by last assertion. You have 4 steps the same. the bug is happen at the first steps but the last steps is successful so maybe it is reason why the report is screen.

**Comment by thephannfq** [comment]
> @alexandrulup
> I think because the result is override by last assertion. You have 4 steps the same. the bug is happen at the first steps but the last steps is successful so maybe it is reason why the report is screen.

I have the same issue with serenity using jbehave, when I use the SerenityParameterizedRunner, if the first/second/third result is failed, and the last result is successful. The serenity report is always GREEN.

@vuthelinh  @wakaleo hope you help me

**Comment by wakaleo** [comment]
Can you provide a sample project?

**Comment by alexandrulup** [comment]
I can confirm that the result is not override by last assertion. Mine is always green, even if the last result is failed.

**Comment by thephannfq** [comment]
> We can't do anything with this without a project that reproduces the issue; I am not seeing it here. Failures anywhere in a parameterised test cause the overall result to fail. @thephannfq - why are you using the parameterised runner _and_ JBehave?

Hi @wakaleo , I will build a small demo and publish it to you soon. I have to use the parameterised runner with JBehave to set up a bdd automation test project, I want to use the same test many times with parameter entered is the browser name (firefox, chrome, safari...)

I use JBehave, Serenity BDD, Browserstack for my project to support testing on multiple browser

**Comment by wakaleo** [comment]
@thephannfq If it is a new project I would advise using Cucumber rather than JBehave. If you want multiple browsers, you can use test contexts much more easily. I can't guarantee the behaviour of combining JBehave and parameterised tests, as both have weird threading models and may not work together.

**Comment by thephannfq** [comment]
Hi @wakaleo 
I built this project for 3 weeks, I couldn't switch to use cucumber
This is my public demo using SerenityParameterizedRunner, in this project, the report is always green though I force the 3rd example case to be failed

https://github.com/thephannfq/serenity-demo

**Comment by wakaleo** [comment]
You cannot combine JBehave with the parameterized runner. They are  not compatible.

**Comment by thephannfq** [comment]
> You cannot combine JBehave with the parameterized runner. They are not compatible.

Do you have any suggestion for me?

**Comment by wakaleo** [comment]
If you are trying to do cross-browser testing, you can use test contexts: https://johnfergusonsmart.com/running-tests-different-environments-browsers-serenity-bdd/

**Comment by thephannfq** [comment]
It can be a good idea. In most cases, I use browserstack for testing multiple browsers, Is there a way to put context value without in maven command?

**Comment by wakaleo** [comment]
Not at the moment, but possibly in a future version.

**Comment by wakaleo** [comment]
For BrowserStack, I just write a script to call maven with different parameters multiple times.

**Comment by phamhiepst** [comment]
> serenity-core-version : 2.0.20
> I'm running data-driven tests with SerenityParameterizedRunner.class and even if the results are failed, I have a green pie chart

I got the same issue when using SoftAssert. Everything is fine if using Assert.

	@Before
	public void setup() {
		softAssert = new SoftAssertions();
	}

	@After
	public void tearDown() {
		softAssert.assertAll();
	}

	@Test
	public void tearDown() {
		softAssert.check1;
		softAssert.check2;
		softAssert.check3; // failure at here
	}

**Comment by wakaleo** [comment]
I’m not sure that the soft assertion will be detected in the @After method, because of the JUnit lifecycle, so this may not be possible.

**Comment by phamhiepst** [comment]
> I’m not sure that the soft assertion will be detected in the @after method, because of the JUnit lifecycle, so this may not be possible.

Hi @wakaleo same issue when moving softAssert.assertAll() into Test.

@Before
public void setup() {
	softAssert = new SoftAssertions();
}

@Test
public void tearDown() {
	softAssert.check1;
	softAssert.check2;
	softAssert.check3; // failure at here
	softAssert.assertAll();
}

I think that current counting failure logic on the Summary page just check if any failure exception happened inside Step block, so it does not care softAssert.assertAll() outside.

**Comment by sortuno** [comment]
> serenity-core-version : 2.0.20
> I'm running data-driven tests with SerenityParameterizedRunner.class and even if the results are failed, I have a green pie chart

I had same issue but now is working properly since I changed the version of the dependency. 

This is how I have it in the pom file.

<dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-cucumber</artifactId>
            <version>1.9.39</version>
            <scope>test</scope>
        </dependency>

**Comment by nambikaas** [comment]
Yes I agree with above point ,if we use 1.9.39 ,issue is resolved but as I need emailable summary report its not available while using any version 2.0.x? Hence pls let me know stable version greater than 2.0.x for getting proper counts in serenity reports

**Comment by wakaleo** [comment]
@nambikaas I'm not sure what issue you are referring to (there are several mentioned in this thread) - can you provide a description of the issue you are facing and a sample project reproducing it?

**Comment by nambikaas** [comment]
@wakaleo Sample project is same like above one https://github.com/thephannfq/serenity-demo
Here we are using Serenity & Rest assured alone , where tests are run using @RunWith(SerenityParameterizedRunner.class)
@UseTestDataFrom(value="$DATADIR/GenericRestRegression_TestData.csv") , 
Issue : Here is even though when test cases are failing , I am getting failing count as 0 in Serenity Summary Report , all test cases are showed as PASSED
![image](https://user-images.githubusercontent.com/60735369/75027044-f7256380-54c3-11ea-9c00-2794e11a5574.png)

**Comment by nambikaas** [comment]
Contd:
And failures are noted only in Full test results  in above screenshot and also pie chart of serenity report available in target/site shows 100% PASSED test cases ,but failures available when click on the link of each test case
And also proper serenity report is generated if i change serenity version to 1.9.45 , but 1.9.45 doesnt support Serenity Summary report for emails

**Comment by eswar2993** [comment]
> Sample pom :
> https://gist.github.com/andreiaionitoaiesv/4095b12cb633a4aa6d672a06399ca6ff
> Sample test:
> https://gist.github.com/andreiaionitoaiesv/18373db4040af50f0a61efdc371e5417
> 
> ![failure](https://user-images.githubusercontent.com/14057245/49863343-c4d09e80-fe08-11e8-8edb-5358e611bf3f.PNG)
> ![landingpage](https://user-images.githubusercontent.com/14057245/49863344-c4d09e80-fe08-11e8-8a6a-ee0dbadf7727.PNG)

Can you please share the code how to make that step assertion failed and show that in the report.
